Src/APIConnection/Exercises.ts

```diff
@@ -1,6 +1,27 @@
-async function fetchExercisesData(exercise: string, appId: string, appKey: string){
+import { User } from '../../Models/User.model'
+import { getAge } from '../UIComponents/utils/utils'
+
+export interface ExerciseItemFromAPI {
+    name: string;
+    met: number;
+    duration_min: number;
+    nf_calories: number;
+}
+
+export interface ExerciseDataFromResponse {
+    exercises: ExerciseItemFromAPI[];
+}
+
+
+async function fetchExercisesData(user: User, exercise: string, appId: string, appKey: string){
     const url = "https://trackapi.nutritionix.com/v2/natural/exercise";
-    const requestJSON = {"query": exercise}
+    const requestJSON = {
+        "query": exercise,
+        "gender": user.gender,
+        "weight_kg": user?.weights[0].weight ?? user.goalWeight, //fallback goalweight just in case the user has not provided us with anything, not sure if possible
+        "height_cm": user.height,
+        "age": getAge(user.dateOfBirth)
```

Comment:
`"age": getAge(user.dateOfBirth) > 0 ? getAge(user.dateOfBirth) : 1`
should resolve issue with api request

Reply:
added, thanks 👍